sphinxgallery/gen_rst.py

```diff
@@ -319,14 +322,17 @@ def make_thumbnail(in_fname, out_fname, width, height):
         import Image
     img = Image.open(in_fname)
     width_in, height_in = img.size
-    scale_w = width / float(width_in)
-    scale_h = height / float(height_in)
+    scale_w = max_width / float(width_in)
+    scale_h = max_height / float(height_in)
 
-    if height_in * scale_w <= height:
+    if height_in * scale_w <= max_height:
         scale = scale_w
     else:
         scale = scale_h
 
+    if scale >= 1.0 and in_fname == out_fname:
+        return
```

Comment:
Just a remark for the future: it's actually a good idea to keep unrelated work in separate branches. I can see that you have the same commit in different PR branches. A good rule of thumb when you start working on a new feature is to do it in a new branch you created off master.